javascript-carousel/main.js

```diff
@@ -0,0 +1,50 @@
+var $carouselImages = document.querySelectorAll('.carousel-image');
+var $circleIcon = document.querySelectorAll('.circle-icon');
+var intervalID = setInterval(toggleRight, 3000);
+var imageIndex = 1;
+var $chevronRight = document.querySelector('.chevron-right');
+var $chevronLeft = document.querySelector('.chevron-left');
+
+function changeImage(index) {
+  if (index <= $carouselImages.length) {
+    for (var i = 0; i < $carouselImages.length; i++) {
+      $carouselImages[i].className = 'carousel-image hidden';
+      $circleIcon[i].className = 'fas fa-circle circle-icon';
+
+      if ((index - 1) === i && index !== 0) {
+        $carouselImages[i].className = 'carousel-image view';
+        $circleIcon[i].className = 'fas fa-circle active circle-icon';
+      }
+
+      if (index === 0) {
+        imageIndex = 1;
+        $carouselImages[0].className = 'carousel-image view';
+        $circleIcon[0].className = 'fas fa-circle active circle-icon';
+      }
```

Comment:
If you make sure that `changeImage()` is always passed a valid index, this code will get much simpler.